### ISSUE TITLE
AnalysisTemplate with DataDog queries does not work properly

### ISSUE THREAD

**Issue by AzgadAGZ** [argoproj/argo-rollouts#3286]
Checklist:

* [x] I've included steps to reproduce the bug.
* [x] I've included the version of argo rollouts.

**Describe the bug**

According to the documentation [here](https://argoproj.github.io/argo-rollouts/analysis/datadog/) and on the code ([datadog.go](https://github.com/argoproj/argo-rollouts/blob/0fab0f2afbd1e7bbd985f76e2ee73bb84a7db6de/metricproviders/datadog/datadog.go#L228) - [datadog_test.go](https://github.com/argoproj/argo-rollouts/blob/0fab0f2afbd1e7bbd985f76e2ee73bb84a7db6de/metricproviders/datadog/datadog_test.go#L125)), I should be able to use the `queries` dict and the `formula` attribute, but when I try to use it is always showing an error that with deeper investigation request the "query" attribute and is not taking the queries inside the AnalysisRun. 

**To Reproduce**

- A DataDog tenant (the free version works)
- rollout.yaml below
- AnalysisTemplate.yaml below

 

### rollout.yaml
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Rollout
metadata:
  name: rollouts-demo
spec:
  replicas: 5
  strategy:
    canary:
      canaryService: rollouts-demo-canary
      stableService: rollouts-demo-stable
      trafficRouting:
        nginx:
          stableIngress: rollouts-demo-stable
      steps:
      - setWeight: 20
      - pause: {duration: 1m}
      - analysis:
          templates:
          - templateName: loq-error-rate
          args:
          - name: service-name
            value: rollout-demo
      - setWeight: 40
      - pause: {duration: 1m}
      - setWeight: 50
      - pause: {duration: 1m}
      - setWeight: 60
      - pause: {duration: 1m}
      - setWeight: 70
      - pause: {duration: 1m}
      - setWeight: 80
      - pause: {duration: 1m}
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: rollouts-demo
  template:
    metadata:
      labels:
        app: rollouts-demo
        tags.datadoghq.com/env: "dev"
        tags.datadoghq.com/service: "rollout-demo"
        tags.datadoghq.com/version: "v1.0.1"

    spec:
      volumes:
        - hostPath:
            path: /var/run/datadog/
          name: apmsocketpath
      containers:
      - name: rollouts-demo
        image: rollout-error-rate:latest
        # image: rollout:latest
        imagePullPolicy: Never
        volumeMounts:
            - name: apmsocketpath
              mountPath: /var/run/datadog
        ports:
        - name: http
          containerPort: 8080
          protocol: TCP
        resources:
          requests:
            memory: 32Mi
            cpu: 5m
        env:
            - name: DD_ENV
              valueFrom:
                fieldRef:
                  fieldPath: metadata.labels['tags.datadoghq.com/env']
            - name: DD_SERVICE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.labels['tags.datadoghq.com/service']
            - name: DD_VERSION
              valueFrom:
                fieldRef:
                  fieldPath: metadata.labels['tags.datadoghq.com/version']
``` 

### AnalysisTemplate.yaml
```yaml
apiVersion: argoproj.io/v1alpha1
kind: AnalysisTemplate
metadata:
  name: loq-error-rate
spec:
  args:
  - name: service-name
  metrics:
  - name: error-rate
    interval: 60s
    initialDelay: 1m
    failureCondition: default(result, 0) >= 0.05
    failureLimit: 3
    count: 5
    provider:
      datadog:
        apiVersion: v2
        interval: 5m
        queries:
          A: sum:trace.http.request.errors{env:prod, service:{{args.service-name}}}.as_count()
          B: sum:trace.http.request.hits{env:prod, service:{{args.service-name}}}.as_count()
        formula: "(A * 100) / B"
``` 

With that configuration, after 1 minute the rollout will start the Analysis, once I describe the AnalysisRun I see the spec is different from what it suppose to be (image below), also is throwing an error since the `query` parameter is empty since I'm not using it.

![image](https://github.com/argoproj/argo-rollouts/assets/78827500/7f5d8133-00d0-4c81-a2e3-e19481d0306e)

**Expected behavior**

**The rollout should use both metrics from DataDog along with the formula parameter to create the result and then evaluate it accordingly.** 

**Screenshots**

![image](https://github.com/argoproj/argo-rollouts/assets/78827500/3104114a-202d-437f-95f4-a5434294a52e)

![image](https://github.com/argoproj/argo-rollouts/assets/78827500/49017892-2635-47c1-a17a-94a29d0fd7a6)

**Version**

> kubectl argo rollouts version
kubectl-argo-rollouts: v1.6.0+7eae71e
  BuildDate: 2023-09-06T18:41:21Z
  GitCommit: 7eae71ed89f1a3769864435bddebe3ca05384df3
  GitTreeState: clean
  GoVersion: go1.20.7
  Compiler: gc
  Platform: darwin/amd64

**Logs**

```
time="2024-01-03T21:11:54Z" level=info msg="Started syncing rollout" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="syncing service" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:11:54Z" level=info msg="Switched selector for service 'rollouts-demo-canary' from '58f9799cd6' to 'dbfcb76bb'" event_reason=SwitchService namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="Found 1 TrafficRouting Reconcilers" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="Reconciling TrafficRouting with type 'Nginx'" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="syncing service" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:11:54Z" level=info msg="Switched selector for service 'rollouts-demo-canary' from 'dbfcb76bb' to '58f9799cd6'" event_reason=SwitchService namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="No changes to canary ingress - skipping patch" ingress=rollouts-demo-rollouts-demo-stable-canary namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="Skipping analysis: isAborted: true, promoteFull: false, rollbackToScaleDownDelay: true, initialDeploy: false" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="Scale down new rs 'rollouts-demo-dbfcb76bb' on abort (30s)" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="New rs 'rollouts-demo-dbfcb76bb' has scaledown deadline annotation: 2024-01-03T21:11:55Z" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="RS 'rollouts-demo-dbfcb76bb' has not reached the scaleDownTime" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="No status changes. Skipping patch" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="Reconciliation completed" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo time_ms=60.302707999999996
time="2024-01-03T21:11:54Z" level=info msg="Started syncing rollout" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="syncing service" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:11:54Z" level=info msg="Switched selector for service 'rollouts-demo-canary' from '58f9799cd6' to 'dbfcb76bb'" event_reason=SwitchService namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="Found 1 TrafficRouting Reconcilers" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:54Z" level=info msg="Reconciling TrafficRouting with type 'Nginx'" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="syncing service" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:11:55Z" level=info msg="Switched selector for service 'rollouts-demo-canary' from 'dbfcb76bb' to '58f9799cd6'" event_reason=SwitchService namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="No changes to canary ingress - skipping patch" ingress=rollouts-demo-rollouts-demo-stable-canary namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Skipping analysis: isAborted: true, promoteFull: false, rollbackToScaleDownDelay: true, initialDeploy: false" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Scale down new rs 'rollouts-demo-dbfcb76bb' on abort (30s)" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="New rs 'rollouts-demo-dbfcb76bb' has scaledown deadline annotation: 2024-01-03T21:11:55Z" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="RS 'rollouts-demo-dbfcb76bb' has reached the scaleDownTime" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Scaled down ReplicaSet rollouts-demo-dbfcb76bb (revision 2) from 1 to 0" event_reason=ScalingReplicaSet namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Not finished reconciling new ReplicaSet 'rollouts-demo-dbfcb76bb'" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Not finished reconciling ReplicaSets" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="No status changes. Skipping patch" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Reconciliation completed" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo time_ms=66.84225
time="2024-01-03T21:11:55Z" level=info msg="Started syncing rollout" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="syncing service" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:11:55Z" level=info msg="Switched selector for service 'rollouts-demo-canary' from '58f9799cd6' to 'dbfcb76bb'" event_reason=SwitchService namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Found 1 TrafficRouting Reconcilers" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Reconciling TrafficRouting with type 'Nginx'" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="syncing service" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:11:55Z" level=info msg="Switched selector for service 'rollouts-demo-canary' from 'dbfcb76bb' to '58f9799cd6'" event_reason=SwitchService namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="No changes to canary ingress - skipping patch" ingress=rollouts-demo-rollouts-demo-stable-canary namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Skipping analysis: isAborted: true, promoteFull: false, rollbackToScaleDownDelay: true, initialDeploy: false" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Scale down new rs 'rollouts-demo-dbfcb76bb' on abort (30s)" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="New rs 'rollouts-demo-dbfcb76bb' has scaledown deadline annotation: 2024-01-03T21:11:55Z" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="RS 'rollouts-demo-dbfcb76bb' has reached the scaleDownTime" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="No status changes. Skipping patch" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Reconciliation completed" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo time_ms=50.236541
time="2024-01-03T21:11:55Z" level=info msg="Started syncing rollout" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="delaying service switch from 58f9799cd6 to dbfcb76bb: ReplicaSet has zero availability" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:11:55Z" level=info msg="Found 1 TrafficRouting Reconcilers" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Reconciling TrafficRouting with type 'Nginx'" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="No changes to canary ingress - skipping patch" ingress=rollouts-demo-rollouts-demo-stable-canary namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Skipping analysis: isAborted: true, promoteFull: false, rollbackToScaleDownDelay: true, initialDeploy: false" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Scale down new rs 'rollouts-demo-dbfcb76bb' on abort (30s)" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="New rs 'rollouts-demo-dbfcb76bb' has scaledown deadline annotation: 2024-01-03T21:11:55Z" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="RS 'rollouts-demo-dbfcb76bb' has reached the scaleDownTime" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Patched: {\"status\":{\"HPAReplicas\":5,\"availableReplicas\":5,\"readyReplicas\":5,\"replicas\":5,\"updatedReplicas\":null}}" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="persisted to informer" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Reconciliation completed" generation=4 namespace=default resourceVersion=1428986 rollout=rollouts-demo time_ms=14.670666
time="2024-01-03T21:11:55Z" level=info msg="Started syncing rollout" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="delaying service switch from 58f9799cd6 to dbfcb76bb: ReplicaSet has zero availability" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:11:55Z" level=info msg="Found 1 TrafficRouting Reconcilers" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Reconciling TrafficRouting with type 'Nginx'" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="No changes to canary ingress - skipping patch" ingress=rollouts-demo-rollouts-demo-stable-canary namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Skipping analysis: isAborted: true, promoteFull: false, rollbackToScaleDownDelay: true, initialDeploy: false" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Scale down new rs 'rollouts-demo-dbfcb76bb' on abort (30s)" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="New rs 'rollouts-demo-dbfcb76bb' has scaledown deadline annotation: 2024-01-03T21:11:55Z" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="RS 'rollouts-demo-dbfcb76bb' has reached the scaleDownTime" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="No status changes. Skipping patch" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Reconciliation completed" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo time_ms=2.940334
time="2024-01-03T21:11:55Z" level=info msg="Started syncing rollout" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="delaying service switch from 58f9799cd6 to dbfcb76bb: ReplicaSet has zero availability" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:11:55Z" level=info msg="Found 1 TrafficRouting Reconcilers" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Reconciling TrafficRouting with type 'Nginx'" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="No changes to canary ingress - skipping patch" ingress=rollouts-demo-rollouts-demo-stable-canary namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Skipping analysis: isAborted: true, promoteFull: false, rollbackToScaleDownDelay: true, initialDeploy: false" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Scale down new rs 'rollouts-demo-dbfcb76bb' on abort (30s)" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="New rs 'rollouts-demo-dbfcb76bb' has scaledown deadline annotation: 2024-01-03T21:11:55Z" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="RS 'rollouts-demo-dbfcb76bb' has reached the scaleDownTime" namespace=default rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="No status changes. Skipping patch" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo
time="2024-01-03T21:11:55Z" level=info msg="Reconciliation completed" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo time_ms=1.2841669999999998
time="2024-01-03T21:18:31Z" level=info msg="Started syncing rollout" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="delaying service switch from 58f9799cd6 to dbfcb76bb: ReplicaSet has zero availability" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:18:31Z" level=info msg="Found 1 TrafficRouting Reconcilers" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="Reconciling TrafficRouting with type 'Nginx'" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="No changes to canary ingress - skipping patch" ingress=rollouts-demo-rollouts-demo-stable-canary namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="Skipping analysis: isAborted: true, promoteFull: false, rollbackToScaleDownDelay: true, initialDeploy: false" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="Scale down new rs 'rollouts-demo-dbfcb76bb' on abort (30s)" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="New rs 'rollouts-demo-dbfcb76bb' has scaledown deadline annotation: 2024-01-03T21:11:55Z" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="RS 'rollouts-demo-dbfcb76bb' has reached the scaleDownTime" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="No status changes. Skipping patch" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="Reconciliation completed" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo time_ms=7.875541999999999
time="2024-01-03T21:18:31Z" level=info msg="syncing service" namespace=default rollout=rollouts-demo service=rollouts-demo-stable
time="2024-01-03T21:18:31Z" level=info msg="syncing service" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:18:31Z" level=info msg="Started syncing rollout" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="delaying service switch from 58f9799cd6 to dbfcb76bb: ReplicaSet has zero availability" namespace=default rollout=rollouts-demo service=rollouts-demo-canary
time="2024-01-03T21:18:31Z" level=info msg="Found 1 TrafficRouting Reconcilers" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="Reconciling TrafficRouting with type 'Nginx'" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="No changes to canary ingress - skipping patch" ingress=rollouts-demo-rollouts-demo-stable-canary namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="Skipping analysis: isAborted: true, promoteFull: false, rollbackToScaleDownDelay: true, initialDeploy: false" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="Scale down new rs 'rollouts-demo-dbfcb76bb' on abort (30s)" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="New rs 'rollouts-demo-dbfcb76bb' has scaledown deadline annotation: 2024-01-03T21:11:55Z" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="RS 'rollouts-demo-dbfcb76bb' has reached the scaleDownTime" namespace=default rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="No status changes. Skipping patch" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo
time="2024-01-03T21:18:31Z" level=info msg="Reconciliation completed" generation=4 namespace=default resourceVersion=1431482 rollout=rollouts-demo time_ms=2.0437089999999998
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by zachaller** [comment]
This ferature has not been released yet it will be available in 1.7

**Comment by zachaller** [comment]
You can see the stable docs do not list the feature: https://argo-rollouts.readthedocs.io/en/stable/analysis/datadog/ however latest which is what github docs uses as well as if you use the drop down will show the feature.
